lambda_code/src/fhir_controller.py

```diff
@@ -99,23 +105,67 @@ def delete_immunization(self, aws_event):
             return self.create_response(500, unhandled_error.to_operation_outcome())
 
     def search_immunizations(self, aws_event) -> dict:
-        params = aws_event["queryStringParameters"]
-
-        if "nhsNumber" not in params:
-            return self._create_bad_request("Query parameter 'nhsNumber' is mandatory")
-        if "diseaseType" not in params:
-            return self._create_bad_request("Query parameter 'diseaseType' is mandatory")
-
-        result = self.fhir_service.search_immunizations(params["nhsNumber"], params["diseaseType"])
+        http_method = aws_event.get("httpMethod")
+        nhsNumberValue = None
+        diseaseTypeValue = None
+        nhsNumberParam = "-nhsNumber"
+        diseaseTypeParam = "-diseaseType"
+
+        if http_method == "POST":
+            # Check if the Content-Type is application/x-www-form-urlencoded
+            content_type = aws_event.get("headers", {}).get("Content-Type")
+            if content_type == "application/x-www-form-urlencoded":
+                # Parse the body of the request
+                body = aws_event["body"]
+                # Decode the base64-encoded body
+                decoded_body = base64.b64decode(body).decode("utf-8")
+                # Parse the decoded body as application/x-www-form-urlencoded
+                parsed_body = urllib.parse.parse_qs(decoded_body)
+                # Access individual form parameters and check if nhsNumber or diseaseType is present
+                nhsNumber_list = parsed_body.get(nhsNumberParam)
+                if nhsNumber_list:
+                    nhsNumberValue = nhsNumber_list[0]
+
+                diseaseType_list = parsed_body.get(diseaseTypeParam)
+                if diseaseType_list:
+                    diseaseTypeValue = diseaseType_list[0]
+                # Continue processing the request
 
+        params = aws_event["queryStringParameters"]
+        # If nhsNumber was not present in body then it should be in params irresepective of GET/POST
+        if nhsNumberValue is None:
```

Comment:
The FHIR docs say that putting the same param in the body and the URL is the same as repeating them, not that one is taken over the other. Which makes sense to me, we can't silently ignore one set of params as they could be different.

In which case, looking at the docs, it's an AND. If they're different it can never return anything as an Immunization can't have 2 ids. If they're the same value then it's the same as having just one. 
I.e. `_id=1 AND _id=2` can never return anything but is a valid call.
`id=1 AND id=1` is just `id=1`.

http://hl7.org/fhir/R4B/http.html#search
http://hl7.org/fhir/R4B/search.html#combining